File: Add Two Numbers - Medium/addtwo.py
from typing import Optional
## 95.23% faster than python3 submission
## 43.23% less mem than python3 submission

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        a = ""
        b = ""
        d = []
        temp = l1 
        temp2 = l2
        while l1!=None or l2!=None:
            if(l1!=None):
                a = str(l1.val)+a
                l1=l1.next
            if(l2!=None):
                b =str(l2.val)+b
                l2=l2.next
        c = str(int(a)+int(b))
        for i in range(0,len(c)):
            d.append(ListNode(int(c[i])))
            if i>0:
                d[i].next = d[i-1]
        return d[-1]

File: 2. Add Two Numbers - Medium/addtwo.js
// Faster than 96.88% online js submission
// Use less mem than 39.68% online js submission
var addTwoNumbers = function(l1, l2) {
    let a = ""
    let b = ""
    let c = []
    while(l1!=null || l2!=null){
        if(l1!=null){
            a = l1.val.toString()+a
            l1=l1.next
        }
        if(l2!=null){
            b = l2.val.toString()+ b
            l2=l2.next
        } 
    }
    let d = (BigInt(a)+BigInt(b)).toString()
    for(let i =0;i<d.length;i++){
        c.push(new ListNode(parseInt(d[i])))
        if(i>0){
            c[i].next = c[i-1]
        }
    }
    return c[c.length-1]
};